lib/ansible/modules/storage/zfs/zpool_facts.py

```diff
@@ -157,11 +160,15 @@ def get_facts(self):
         cmd = [self.module.get_bin_path('zpool')]
 
         cmd.append('get')
-        cmd.append('-H')
+        if not self.is_sunos:
+            # solaris 11 does not support -H
+            cmd.append('-H')
         if self.parsable:
             cmd.append('-p')
-        cmd.append('-o')
-        cmd.append('name,property,value')
+        if self.is_freebsd:
```

Comment:
-H/-o are both supported on Linux: https://github.com/zfsonlinux/zfs/blob/master/cmd/zpool/zpool_main.c#L7686-L7689. @sysadmin75 which platform do you want to support?